lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -552,15 +552,20 @@ def confirm_cluster_reachable
     end
 
     def confirm_namespace_exists
```

Comment:
As part of this PR I looked over where we use `Kubectl#run`, and I noticed that this file requests namespaces twice and implements retries itself instead of using the built-in mechanism.